src/universal/releaseFlags.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Release Flags are feature flags that are immutable for a given release.
+ *
+ * They're exposed to the client via webpack.DefinePlugin, which allows us to
+ * take advantage of UglifyJS' dead code eliminator.  If you want to use a
+ * release flag on the client, DO NOT IMPORT from this module.  Instead,use
+ * the global RELEASE_FLAGS.<featureName>.  To take advantage of DCE, you
+ * will have to directly pass the name to a control structure, rather than passing
+ * it to a function. e.g.
+ *
+ *   <div>
+ *     {RELEASE_FLAGS.newThing && <NewThing />}
+ *     {RELEASE_FLAGS.foo ? <Foo /> : <Bar />}
+ *   </div>
+ *
+ * They're exposed to the GraphQL server via context.releaseFlags
+ *
+ * @flow
+ */
+
+export default {};
```

Comment:
i always prefer an array or Set opposed to an object full of `true` values because it means i can't sneak in antipatterns later, but i'm cool with either one.

Reply:
> it means i can't sneak in antipatterns later

Can you clarify on this? I don't really know what you're referring to.

Remember that this module is not imported in the client app. It's used by webpack at build-time to replace occurrences of `__RELEASE_FLAGS__.aCoolFeature` with e.g. `true` or `false` or `"blahblahblah"`.

Reply:
Bingo! It's that "blahblahblah" that i'm worried about. It's a flag, so it should be a boolean.

I propose:
export default [__RELEASE_FLAGS__A_COOL_FEATURE, __FEATURE_TWO__];

Then, in webpack:
```
releaseFlags.reduce((obj, flag) => obj[flag] = true; return obj, {})
```

Reply:
I think any JS type is perfectly valid for a release flag value!

Reply:
well sure, but then it shouldn't be called a flag :wink: